docs/products/access/api-reference/custom-registration/custom-registration-v1.md

```diff
@@ -0,0 +1,149 @@
+# Custom Registration v1
+
+## Authentication
+
+These endpoints require HTTP Basic Authentication with the Client Credentials of the Client that
+triggers the Custom Registration.
+
+## Init Request
+
+The Init Step is only used for the TWO_STEP flow. The ONE_STEP flow uses the Complete Step.
+
+Endpoint: `POST /oauth/custom-registration/{idp}/init`
+
+| Parameter | Description                  |
+|-----------|------------------------------|
+| `idp`     | Identity provider identifier |
+
+JSON body parameters:
+
+| Param   | Required  | Description                                                                          |
+|---------|-----------|--------------------------------------------------------------------------------------|
+| `data`  | no        | Raw registration request data which will be provided to the Onegini Extension Engine |
+
+Example request:
+
+```http
+POST /oauth/custom-registration/example-custom-registration-idp/init HTTP/1.1
+Host: onegini.example.com
+Content-Type: application/json
+Authentication: Basic Y2xpZW50OnNlY3JldA==
+
+{
+    "data": "{\"custom_json_key\":\"custom json data\"}" // optional
+}
+```
+
+Example success response:
+
+```http
+HTTP/1.1 200 Ok
+Content-Type: application/json;charset=UTF-8
+Cache-Control: no-store
+Pragma: no-cache
+
+{
+   "transaction_id": "123123",   //something unique, should be passed with complete step
+   "data": "12349876",           // e.g. a challenge code.
+   "status": 2000
+}
+```
+
+In the event of an error in the Access Service, one of the following [error codes](#server-error-codes) will be returned.
+
+It is up to the scripts executed by the XE to determine if the request was successful or not when everything looks fine for the Access
```

Comment:
Should we use XE? Or write it out as Extenstion Engine?

Reply:
I'll write it out fully. Already did that in other places.